src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,161 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+
+    private static RegistrationServiceImpl registrationService;
+    private static StorageDao storageDao;
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Test
+    void register_validUser() {
+        User validUser = new User();
+        validUser.setAge(20);
+        validUser.setPassword("passwordValid");
+        validUser.setLogin("loginValid");
+
+        registrationService.register(validUser);
+        assertNotNull(storageDao.get("loginValid"));
+        assertEquals(validUser, storageDao.get("loginValid"));
```

Comment:
It's important to also verify that the storage contains only one user after the registration. This will ensure that the `register` method doesn't add the same user multiple times.